sys/dev/misc/efirt/efidev.c

```diff
@@ -77,9 +77,24 @@ efidev_ioctl(struct dev_ioctl_args *ap)
 	case EFIIOC_GET_TABLE:
 	{
 		struct efi_get_table_ioc *egtioc =
-		    (struct efi_get_table_ioc *)addr;
+			(struct efi_get_table_ioc *)addr;
+		struct efi_esrt_table *esrt = NULL;	
 
-		error = efi_get_table(&egtioc->uuid, &egtioc->ptr);
+		error = efi_get_table(&egtioc->uuid, (void **)&esrt);
```

Comment:
You probably want to `break` if `error != 0` here rather than continuing on wrong UUID.

Does `efi_get_table()` allocate the memory it assigns to `esrt`? If so, the copy should be freed on all code paths (with `goto gt_out` like in other cases below).

Reply:
@SergiiDmytruk it is not `malloc`-ed, so should not be freed with `free`, correct? that code generates segfault and kernel crash

Reply:
I see, it's just a result of `PHYS_TO_DMAP`.